torch/legacy/nn/ClassNLLCriterion.py

```diff
@@ -27,24 +27,28 @@ def updateOutput(self, input, target):
             self.sizeAverage,
             self.weights,
             self.total_weight_tensor,
-            self.ignore_index
+            self.ignore_index,
+            True,  # reduce
         )
         self.output = self.output_tensor[0]
         return self.output
 
     def updateGradInput(self, input, target):
         self.gradInput.resize_as_(input).zero_()
         target = target.long()
+        implicit_gradOutput = torch.ones(1).type_as(input)
```

Comment:
torch.ones_like(input)

Reply:
The implicit gradOutput should be a tensor with one element, the value 1 inside it, so I don't think `torch.ones_line(input)` is what I'm looking for here. Is there a nicer way to write this?

Reply:
My mistake -- this is fine

Reply:
Actually I think this won't work for CUDA

Reply:
How so? Do you mean that if input is CUDA then implicit_gradOutput won't be?

Reply:
Um, sorry it should work. However I just realized that it might not respect the device of input (which is a bug in type_as)